### ISSUE TITLE
[Bug] Invalid URI: The hostname could not be parsed. Error with WithMultiCloudSupport(true) and OS account in AcquireTokenSilent

### ISSUE THREAD

**Issue by NakulSabharwal** [AzureAD/microsoft-authentication-library-for-dotnet#3769]
**Logs and network traces**
```
Invalid URI: The hostname could not be parsed.

   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at System.UriBuilder.get_Uri()
   at Microsoft.Identity.Client.AuthorityInfo.AuthorityInfoHelper.CreateAuthorityWithEnvironment(AuthorityInfo authorityInfo, String environment)
   at Microsoft.Identity.Client.AuthorityInfo.AuthorityInfoHelper.<CreateAuthorityForRequestAsync>d__1.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Identity.Client.Instance.Authority.<CreateAuthorityForRequestAsync>d__4.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Identity.Client.Internal.Requests.Silent.SilentRequest.<UpdateRequestWithAccountAsync>d__10.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Identity.Client.Internal.Requests.Silent.SilentRequest.<ExecuteAsync>d__5.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Identity.Client.Internal.Requests.RequestBase.<RunAsync>d__12.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Identity.Client.ApiConfig.Executors.ClientApplicationBaseExecutor.<ExecuteAsync>d__2.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
```

**Which version of MSAL.NET are you using?**
MSAL.NET 4.47.2

**Platform**
.NET 4.7

**What authentication flow has the issue?**
* Desktop
    * Silent auth/auth-login
    * Integrated Windows Authentication (WAM)

**Is this a new or existing app?**
a. The app is in production, and I have upgraded to a new version of MSAL.

**Repro**

```csharp
            return PublicClientApplicationBuilder.Create(clientId)
                  .WithDefaultRedirectUri()
                  .WithAuthority(AzureCloudInstance.AzurePublic, AadAuthorityAudience.AzureAdAndPersonalMicrosoftAccount)
                  .WithWindowsBroker()
                  .WithMsalLogging()
                  .WithMultiCloudSupport(true);
```

Then try to use auto-login
```csharp

            AuthenticationResult authResult
                = await authPublicClientApp.AcquireTokenSilent(scopes, account)
                        .WithCorrelationId(AppUtils.CorrelationId)
                        .ExecuteAsync();

```

**Expected behavior**
Auto-login was working fine before enabling WithMultiCloudSupport(true) but after enabling it auto-login is not working.

**Actual behavior**
Auto-login is not working with WAM with WithMultiCloudSupport(true)

**Possible solution**
There seems to be issue with URL parsing in MSAL library

**Comment by NakulSabharwal** [comment]
@pmaytak This bug is a P0 for our team, because we are blocked and cannot update MSAL and not able to use WithMultiCloudSupport, its not working even for our public cloud users and special cloud users as well.

**Comment by pmaytak** [comment]
@NakulSabharwal I wasn't able to repro with a public cloud user.
Can you post code snippets of:
- how you call acquire token interactive
- how do you get an account that you pass into the AcquireTokenSilent

For the account that you pass into AcquireTokenSilent, what is the `account.Environment` property? Does it look properly formatted, something like `login.windows.net`?

**Comment by bgavrilMS** [comment]
Can you also send some verbose logs our way?

**Comment by NakulSabharwal** [comment]
@pmaytak 
#### To auto-login we do the following :
#### We use the following WAM configuration
```csharp
string  authority = "https://login.windows.net/";
string tenant = "common";

PublicClientApplicationBuilder.Create(clientId)
.WithDefaultRedirectUri()
.WithAuthority(authority, tenant)
.WithWindowsBroker()
.WithMultiCloudSupport(true)
.WithMsalLogging()
.Build();
```
#### then we use the following code to select the os account
```csharp
IAccount account = PublicClientApplication.OperatingSystemAccount;
```
#### then use the following code to acquire token silently using account
```csharp
AuthenticationResult authResult = await authPublicClientApp.ActivePublicClientApp?.AcquireTokenSilent(scopes, account)
.WithCorrelationId(AppUtils.CorrelationId)
.ExecuteAsync();
```
@bgavrilMS I have sent full verbose logs using Teams.

**Comment by gladjohn** [comment]
@NakulSabharwal this is fixed in MSAL 4.48.0
